artifactory/services/buildinfo.go

```diff
@@ -87,3 +95,37 @@ func (bis *BuildInfoService) PublishBuildInfo(build *buildinfo.BuildInfo, projec
 	log.Debug("Artifactory response:", resp.Status)
 	return summary, nil
 }
+
+func (bis *BuildInfoService) DeleteBuildInfo(build *buildinfo.BuildInfo, projectKey string, buildNumberFrequency int) error {
+	params := CreateDeleteBuildInfoBody(build, projectKey, buildNumberFrequency)
+	content, err := json.Marshal(params)
+	if err != nil {
+		return err
+	}
+	httpClientsDetails := bis.GetArtifactoryDetails().CreateHttpClientDetails()
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+	resp, body, err := bis.client.SendPost(bis.GetArtifactoryDetails().GetUrl()+"api/build/delete", content, &httpClientsDetails)
+	if err != nil {
+		log.Error("Error occurred while deleting build info", err)
```

Comment:
I suggest to return a new error with this prefix and the original error message instead of logging (which will display multiple error messages in the log)